src/main/java/de/unijena/cheminf/fragment/fingerprint/CountFingerprint.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Betuel Sevindik, Felix Baensch, Jonas Schaub, Christoph Steinbeck, and Achim Zielesny
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+package de.unijena.cheminf.fragment.fingerprint;
+
+import org.openscience.cdk.fingerprint.ICountFingerprint;
+
+import java.util.HashMap;
+
+/**
+ * The CountFingerprint class implements the CDK interface ICountFingerprint.
+ * ICountFingerprint provides useful methods to obtain information about the calculated count fingerprint.
+ */
+public class CountFingerprint implements ICountFingerprint {
+    //<editor-fold desc="private  class variables" defaultstate="collapsed">
+    /**
+     * Is an array containing all predefined unique SMILES.
+     */
+    private String[] fragmentArrayOfUniqueSmiles;
+    /**
+     * The HashMap maps the unique SMILES to the position they have in the array.
+     */
+    private HashMap<String,Integer> uniqueSmilesToPositionMap;
+    /**
+     * Map
+     */
+    private HashMap<Integer, Integer> rawMap;
```

Comment:
These three variables could be final and should maybe be final because the underlying set of fragments is only assigned once and should not be changed later, right?

Reply:
And "Map" is an insufficient description